app/models/package_manager/packagist.rb

```diff
@@ -52,8 +52,10 @@ def self.recent_names
     end
 
     def self.project(name)
-      get("https://packagist.org/packages/#{name}.json")
-        &.fetch("package")
+      # The main v2 endpoint excludes dev versions, so if that list
+      # of versions is empty, fallback to the dev list of versions.
+      get("https://repo.packagist.org/p2/#{name}.json").dig("packages", name).presence || 
```

Comment:
Do we need to encode the name in here or is it already coming in encoded?

Reply:
It looks like we don't do any URI escaping for any of the PackageManager::Base subclasses (except for a one-off slash-escaping in NPM). Bad URIs would get raised as `URI::InvalidURIError`, and we do have a small flow of those, but it doesn't seem like a big issue (ruby URI parsing vulnerabilities notwithstanding).